server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -954,13 +965,104 @@ protected void configure() {
                 bind(supportAction).asEagerSingleton();
             }
         }
+
+        // register dynamic ActionType -> transportAction Map used by NodeClient
+        if (FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS)) {
+            bind(DynamicActionRegistry.class).toInstance(dynamicActionRegistry);
+        }
     }
 
     public ActionFilters getActionFilters() {
         return actionFilters;
     }
 
+    @SuppressWarnings("unchecked")
+    public DynamicActionRegistry<? extends ActionRequest, ? extends ActionResponse> getDynamicActionRegistry() {
+        return dynamicActionRegistry;
+    }
+
     public RestController getRestController() {
         return restController;
     }
+
+    /**
+     * The DynamicActionRegistry maintains a registry mapping {@link ExtensionAction} instances to {@link TransportAction} instances.
+     * <p>
+     * This class is modeled after {@link NamedRegistry} but provides both register and unregister capabilities.
+     *
+     * @opensearch.internal
+     */
+    public static class DynamicActionRegistry<Request extends ActionRequest, Response extends ActionResponse> {
```

Comment:
Do these generics do anything? Looks like in its usage such as NodeClient they aren't provided

Reply:
I have been fighting with those generics for at least an hour.  Without them, the `get()` call in NodeClient fails.

If you know how to properly do them or can point me to a "Generics 101" tutorial, I'd buy you a green beverage of choice.

Reply:
Things work when I keep the maps separate but trying to have a combined get method that includes subclasses is not.